src/mem3/src/mem3_shards.erl

```diff
@@ -249,6 +269,21 @@ code_change(_OldVsn, #st{}=St, _Extra) ->
 
 %% internal functions
 
+start_changes_listener(SinceSeq) ->
```

Comment:
these references confused me a bit...but if iiuc, we're spawning a separate process to check if the mem3_shards process died, and once it does, we terminate the listen_for_changes process that was spawned by mem3_shards? If that's the case, what's the purpose behind this? Before we were just monitoring it.